From c5ff3c36f888071eebd8f6fdaaa9a2243f87366b Mon Sep 17 00:00:00 2001
From: ghoker143 <ghoker51@protonmail.com>
Date: Fri, 23 Oct 2026 17:54:08 +0800
Subject: [PATCH 3/3] tools: build: fix missing ccache dependency

Conditionally-built host tools lack dependency on ccache/compile, potentially leading to parallel build failures. (Rebased from the qosmio-based tree; the bug is still present here.)
---
 tools/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/Makefile b/tools/Makefile
index 79d3954fef..69e6a0f579 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -156,7 +156,7 @@ else
 endif
 
 ifneq ($(CONFIG_CCACHE)$(CONFIG_SDK),)
-$(foreach tool, $(filter-out zstd zlib xz pkgconf patch ninja meson libressl expat cmake,$(tools-y)), $(eval $(curdir)/$(tool)/compile += $(curdir)/ccache/compile))
+$(foreach tool, $(filter-out zstd zlib xz pkgconf patch ninja meson libressl expat cmake,$(sort $(tools-y) $(tools-))), $(eval $(curdir)/$(tool)/compile += $(curdir)/ccache/compile))
 tools-y += ccache xxhash
 $(curdir)/xxhash/compile := $(curdir)/cmake/compile
 $(curdir)/ccache/compile := $(curdir)/xxhash/compile
-- 
2.56.0

